File: Ejemplos/Comunicacion/Python/FifoA.py
import sys
import os

try:
    os.mkfifo("./fifo", 0o600)  # Crear el FIFO si no existe
except OSError as err:
    print(f"Error al crear fifo: {err}")
    sys.exit(1)

try:
    fifo = os.open("./fifo", os.O_WRONLY)  # Abrir el FIFO para escribir
    os.write(fifo, "Café La Morenita".encode())  # Escribir en el FIFO
    os.close(fifo)  # Cerrar el FIFO
    os.unlink("./fifo")  # Eliminar el FIFO
except OSError as err:
    print(f"Error al trabajar con el FIFO: {err}")
    sys.exit(1)

File: Ejemplos/Comunicacion/C/fifoA.c
#include<stdlib.h>
#include<stdio.h>
#include<fcntl.h>
#include<unistd.h>
#include<string.h>
#include<sys/stat.h>

struct persona
{
    int dni;
    char nombre[30];	
};

int main (int argc, char *argv[])
{
    if (mkfifo("./fifo",0666) == -1)
    {
        perror("Error al crear ./fifo");
        return EXIT_FAILURE;	
    }

    int w = open("./fifo", O_WRONLY);	

    struct persona alumno;
    alumno.dni = 29383491;
    strcpy(alumno.nombre,"Darío Hirschfeldt");
    if (write(w, &alumno, sizeof(struct persona)) == -1)
    {
        perror("Error en write()");
        return EXIT_FAILURE;
    }	

    close(w);

    unlink("./fifo");

    return EXIT_SUCCESS;
}
